### ISSUE TITLE
Two orders placed, one payment sent, two orders marked as paid

### ISSUE THREAD

**Issue by Wegerich** [WeProgramIT/cryptowoo-monero-addon#15]
A customer placed one order on mobile device using monero. After 2 minutes they placed an different order on a desktop device using monero.

They sent payment for the desktop order to the requested address. 

Both orders were marked as paid by the plugin, even though only one payment was sent to one address.

**Comment by Olsm** [comment]
@Wegerich I have created and tested successfully a fix for this in commit ec7cc8a51c5010a3723997b5c44ab743950ac456. Maybe you can do some quick testing also and we can push this update out more quickly. Thanks. 

BTW this should also improve issue #9 by making it faster to scan blocks for txs so more blocks can be checked each cron job.

**Comment by Wegerich** [comment]
I have changed the version on the site to use this new version. Do you want to try placing two orders for this £0.01 product? https://www.reagent-tests.uk/product/test/

**Comment by Wegerich** [comment]
That seems to have fixed it

**Comment by Olsm** [comment]
Yes. I made two orders. First one on mobile then one on desktop. The amount of Monero was exactly the same in both orders. Then I sent payment to the second order, which completed. The first order did not complete. Worked exactly like expected. This is exactly how I wanted the integrated address integration to work. Thanks to @flxstn I figured out the issue with decrypting the payment id and this solution both fix this issue as well as improve performance and quantity of txs/blocks scanned in cron jobs.
